### DISCUSSION TITLE
Cannot update directory structure in prompt

### DISCUSSION THREAD

**Discussion by angoodkind** [ohmyzsh/ohmyzsh#10322] [Help] [answered]
I am using the `oh-my-zsh` theme `jtriley`. The theme code, reproduced below, displays the entire directory path:

    PROMPT="%{$fg_bold[cyan]%}%T%{$fg_bold[green]%} %{$fg_bold[green]%}%d
    %{$fg_bold[yellow]%}%% %{$reset_color%}"

I know that there are ways to change the displayed directories, using the `prompt_dir()` function,  following instructions like [these](https://shandou.medium.com/how-to-shorten-zsh-prompt-oh-my-zsh-14185f3e7ab7). However, when I insert this below the code above, I do not get a change.

How can I alter a theme file, in order to only show 2 parent directories?

**Comment by mcornella** [comment, marked as answer]
The `prompt_dir` function hack only works for the agnoster theme. For `jtriley` you need to modify the `PROMPT` variable directly: [the `%d` part is what gets expanded to the full directory](https://zsh.sourceforge.io/Doc/Release/Prompt-Expansion.html#Shell-state), you can set it to display up to 2 parents you can use `%3d` (the 3 means display 3 components of the path). Have a look at the [Prompt Expansion docs](https://zsh.sourceforge.io/Doc/Release/Prompt-Expansion.html#Shell-state) to learn more about what these % sequences mean.

You can either set the `PROMPT` variable in your zshrc file, after OMZ has loaded, or copy the theme file to `$ZSH_CUSTOM/themes/` and modify the line in that file (see [customization](https://github.com/ohmyzsh/ohmyzsh/wiki/Customization#overriding-and-adding-themes)).

**Reply by angoodkind** [reply]
Thanks for the explanation!
